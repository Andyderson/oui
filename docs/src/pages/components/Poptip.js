
  import React from 'react';
  import ComponentContent from '../../layouts/templates/ComponentContent';

   import data from '../../../../data/components/Poptip/react.json'; import examples from '../../../../src/components/Poptip/example'; import readme from 'raw-loader!../../../../src/components/Poptip/README.md'; import sassExamples from '../../../../data/components/Poptip/sass.json';

  let dataObject = {
    react: data ? data : null, 
    examples: examples ? examples : null,
    readme: readme ? readme : null,
    sass: sassExamples ? sassExamples.example : null,
  }
  
  class PoptipComponent extends React.Component {
  
    render() {
      return (
        <ComponentContent data={ dataObject } />
      );
    };
  }
  
  export default PoptipComponent
