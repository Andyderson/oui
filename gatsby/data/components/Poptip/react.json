{"description":"Displays help text when hovering on an element.","displayName":"Poptip","methods":[],"props":{"children":{"type":{"name":"node"},"required":true,"description":"Content that, when hovered on, makes the Poptip appear"},"content":{"type":{"name":"string"},"required":true,"description":"Text that appears within the poptip"},"horizontalAttachment":{"type":{"name":"enum","value":[{"value":"'left'","computed":false},{"value":"'center'","computed":false},{"value":"'right'","computed":false}]},"required":false,"description":"Side of the poptip that should attach to the `children`"},"horizontalTargetAttachment":{"type":{"name":"enum","value":[{"value":"'left'","computed":false},{"value":"'center'","computed":false},{"value":"'right'","computed":false}]},"required":false,"description":"Side of `children` that should attach to the poptip"},"testSection":{"type":{"name":"string"},"required":false,"description":"Hook for automated JavaScript tests"},"verticalAttachment":{"type":{"name":"enum","value":[{"value":"'top'","computed":false},{"value":"'middle'","computed":false},{"value":"'bottom'","computed":false}]},"required":false,"description":"Vertical edge of the poptip that should touch the `children`"},"verticalTargetAttachment":{"type":{"name":"enum","value":[{"value":"'top'","computed":false},{"value":"'middle'","computed":false},{"value":"'bottom'","computed":false}]},"required":false,"description":"Vertical edge of the `children` that should touch the poptip"}},"private":false}