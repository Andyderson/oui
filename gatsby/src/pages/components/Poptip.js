
  import React from 'react';
  import ComponentContent from '../../layouts/templates/ComponentContent';

   import examples from '../../../../src/components/Poptip/example'; import readme from 'raw-loader!../../../../src/components/Poptip/README.md';

  const sassData = {"name":"Poptip\n","description":"Information bubbles on hover.","start":{"line":1,"column":1},"end":{"line":82,"column":4},"example":[{"type":"html","description":null,"code":"<div style=\"position: relative; min-height: 2rem\">\n  <div class=\"oui-pop--tip\" style=\"opacity: 1;\">\n    This is a pop-tip. Use only limited text.\n  </div>\n</div>"},{"type":"html","description":null,"code":"<div style=\"position: relative; min-height: 2rem\">\n  <div class=\"oui-pop--tip oui-pop--tip--arrow-top-left\" style=\"opacity: 1;\">\n    This is a pop-tip. Use only limited text.\n  </div>\n</div>"},{"type":"html","description":null,"code":"<div style=\"position: relative; min-height: 2rem\">\n  <div class=\"oui-pop--tip oui-pop--tip--arrow-top-center\" style=\"opacity: 1;\">\n    This is a pop-tip. Use only limited text.\n  </div>\n</div>"},{"type":"html","description":null,"code":"<div style=\"position: relative; min-height: 2rem\">\n  <div class=\"oui-pop--tip oui-pop--tip--arrow-top-right\" style=\"opacity: 1;\">\n    This is a pop-tip. Use only limited text.\n  </div>\n</div>"},{"type":"html","description":null,"code":"<div style=\"position: relative; min-height: 2rem\">\n  <div class=\"oui-pop--tip oui-pop--tip--arrow-right-top\" style=\"opacity: 1;\">\n    This is a pop-tip. Use only limited text.\n  </div>\n</div>"},{"type":"html","description":null,"code":"<div style=\"position: relative; min-height: 2rem\">\n  <div class=\"oui-pop--tip oui-pop--tip--arrow-right-center\" style=\"opacity: 1;\">\n    This is a pop-tip. Use only limited text.\n  </div>\n</div>"},{"type":"html","description":null,"code":"<div style=\"position: relative; min-height: 2rem\">\n  <div class=\"oui-pop--tip oui-pop--tip--arrow-right-bottom\" style=\"opacity: 1;\">\n    This is a pop-tip. Use only limited text.\n  </div>\n</div>"},{"type":"html","description":null,"code":"<div style=\"position: relative; min-height: 2rem\">\n  <div class=\"oui-pop--tip oui-pop--tip--arrow-bottom-right\" style=\"opacity: 1;\">\n    This is a pop-tip. Use only limited text.\n  </div>\n</div>"},{"type":"html","description":null,"code":"<div style=\"position: relative; min-height: 2rem\">\n  <div class=\"oui-pop--tip oui-pop--tip--arrow-bottom-center\" style=\"opacity: 1;\">\n    This is a pop-tip. Use only limited text.\n  </div>\n</div>"},{"type":"html","description":null,"code":"<div style=\"position: relative; min-height: 2rem\">\n  <div class=\"oui-pop--tip oui-pop--tip--arrow-bottom-left\" style=\"opacity: 1;\">\n    This is a pop-tip. Use only limited text.\n  </div>\n</div>"},{"type":"html","description":null,"code":"<div style=\"position: relative; min-height: 2rem\">\n  <div class=\"oui-pop--tip oui-pop--tip--arrow-left-bottom\" style=\"opacity: 1;\">\n    This is a pop-tip. Use only limited text.\n  </div>\n</div>"},{"type":"html","description":null,"code":"<div style=\"position: relative; min-height: 2rem\">\n  <div class=\"oui-pop--tip oui-pop--tip--arrow-left-center\" style=\"opacity: 1;\">\n    This is a pop-tip. Use only limited text.\n  </div>\n</div>"},{"type":"html","description":null,"code":"<div style=\"position: relative; min-height: 2rem\">\n  <div class=\"oui-pop--tip oui-pop--tip--arrow-left-top\" style=\"opacity: 1;\">\n    This is a pop-tip. Use only limited text.\n  </div>\n</div>"}]};
  const reactData = {"description":"Displays help text when hovering on an element.","displayName":"Poptip","methods":[],"props":{"children":{"type":{"name":"node"},"required":true,"description":"Content that, when hovered on, makes the Poptip appear"},"content":{"type":{"name":"string"},"required":true,"description":"Text that appears within the poptip"},"horizontalAttachment":{"type":{"name":"enum","value":[{"value":"'left'","computed":false},{"value":"'center'","computed":false},{"value":"'right'","computed":false}]},"required":false,"description":"Side of the poptip that should attach to the `children`"},"horizontalTargetAttachment":{"type":{"name":"enum","value":[{"value":"'left'","computed":false},{"value":"'center'","computed":false},{"value":"'right'","computed":false}]},"required":false,"description":"Side of `children` that should attach to the poptip"},"testSection":{"type":{"name":"string"},"required":false,"description":"Hook for automated JavaScript tests"},"verticalAttachment":{"type":{"name":"enum","value":[{"value":"'top'","computed":false},{"value":"'middle'","computed":false},{"value":"'bottom'","computed":false}]},"required":false,"description":"Vertical edge of the poptip that should touch the `children`"},"verticalTargetAttachment":{"type":{"name":"enum","value":[{"value":"'top'","computed":false},{"value":"'middle'","computed":false},{"value":"'bottom'","computed":false}]},"required":false,"description":"Vertical edge of the `children` that should touch the poptip"}},"private":false};

  let dataObject = {
    react: reactData, 
    examples: examples ? examples : null,
    readme: readme ? readme : null,
    sass: sassData,
  }
  
  class PoptipComponent extends React.Component {
  
    render() {
      return (
        <ComponentContent data={ dataObject } />
      );
    };
  }
  
  export default PoptipComponent
