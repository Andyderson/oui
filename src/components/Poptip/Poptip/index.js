import React from 'react';

/**
 * @param {Object} props - Properties passed to component
 * @returns {ReactElement}
 * @private
 */
const Poptip = (props) => (
  <div
    className={ 'oui-pop--tip' }
    style={{ display: 'inline-block', opacity: 1, position: 'initial' }}
    data-test-section={ props.testSection }>
    { props.children }
  </div>
);

Poptip.propTypes = {
  /** Content that appears within the popover body */
  children: React.PropTypes.string.isRequired,
  /** Hook for automated JavaScript tests */
  testSection: React.PropTypes.string,
};

export default Poptip;
