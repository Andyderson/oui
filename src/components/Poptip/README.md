# Poptip Component

- Add storybook link 
- Add design guidelines