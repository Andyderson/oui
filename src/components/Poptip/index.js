import React from 'react';
import PropTypes from 'prop-types';

import { Tooltip } from 'react-tippy';

/**
 * Displays help text when hovering on an element.
 * @param {Object} props - Properties passed to component
 * @returns {ReactElement}
 */

const Poptip = ({
  children,
  content,
  disable,
  position,
  trigger,
  theme,
}) => (
  <Tooltip
    arrow={ true }
    disable={ disable }
    html={ <div>{ content }</div> }
    position={ position }
    theme={ theme }
    trigger={ trigger }>
    <div>{ children }</div>
  </Tooltip>
);

Poptip.displayName = 'Poptip';

Poptip.propTypes = {
  /** Content that, when hovered on, makes the Poptip appear */
  children: PropTypes.node.isRequired,
  /** content could be a node of a string */
  content: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.string,
  ]),
  /** disable value | default => false */
  disable: PropTypes.bool,
  /** position value options | default => top */
  position: PropTypes.oneOf([
    'top',
    'bottom',
    'left',
    'right',
  ]),
  /** background color | default => dark */
  theme: PropTypes.string,
  /** trigger value options | default => mouseenter focus */
  trigger: PropTypes.oneOf([
    'mouseenter',
    'focus',
    'click',
    'manual',
  ]),
};

export default Poptip;
