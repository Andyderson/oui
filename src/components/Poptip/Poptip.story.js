import React from 'react';

import { storiesOf, addDecorator } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import  { withKnobs, text, boolean, number, select } from '@storybook/addon-knobs';
import { WithNotes } from '@storybook/addon-notes';

import Poptip from './index.js';
import Button from '../Button';

const stories = storiesOf('Poptip', module);
stories
  .addDecorator(withKnobs)
  .addDecorator(story => (
    <div id="root-preview">
        <div className="flexboxCenter">
            {story()}
        </div>
    </div>
  ));

stories
  .addWithInfo('basic poptip', () => <div className="position--relative height--100">
    <Poptip className="text--center" content="Ipsa officiis bad-news minus earum a aperiam! Aperiam reiciendis vitae nihil libero et, hic!">
        <Button width="default">
        Basic Poptip
        </Button>
    </Poptip>
    </div>)
    .addWithInfo('poptip to the right', () => <div className="position--relative height--100"> 
        <Poptip
            content="Ipsa officiis bad-news minus earum a aperiam! Aperiam reiciendis vitae nihil libero et, hic!"
            horizontalAttachment="left"
            horizontalTargetAttachment="right"
            verticalAttachment="middle"
            verticalTargetAttachment="middle">
            <Button style="unstyled" width="default">
                <a>Poptip on the right</a>
            </Button>
        </Poptip>
    </div>)
    .addWithInfo('poptip with advanced positioning', () => <div className="position--relative height--100">
        <Poptip
            content="Ipsa officiis bad-news minus earum a aperiam! Aperiam reiciendis vitae nihil libero et, hic!"
            horizontalAttachment="left"
            horizontalTargetAttachment="left"
            verticalAttachment="top"
            verticalTargetAttachment="bottom">
            <Button width="default">
                Poptip With Advanced Positioning
            </Button>
        </Poptip>
    </div>);
