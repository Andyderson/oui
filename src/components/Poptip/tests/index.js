import React from 'react';
import Poptip from '../index';
import { shallow } from 'enzyme';

describe('components/Poptip', () => {
  it('should render an `OverlayWrapper` with correct props', () => {
    const component = shallow(
      <Poptip
        content="goose"
        horizontalAttachment="center"
        horizontalTargetAttachment="center"
        verticalAttachment="middle"
        verticalTargetAttachment="middle"
        testSection="heeeeeloooo">
        <button>duck</button>
      </Poptip>
    );

    const OverlayWrapper = component.find('OverlayWrapper');

    expect(OverlayWrapper.length).toBe(1);
    expect(OverlayWrapper.prop('behavior')).toBe('hover');
    expect(OverlayWrapper.prop('overlay').props.children).toBe('goose');
    expect(OverlayWrapper.prop('overlay').props.testSection).toBe('heeeeeloooo');
    expect(OverlayWrapper.prop('children').type).toBe('button');
  });
});
