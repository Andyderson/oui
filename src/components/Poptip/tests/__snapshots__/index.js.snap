// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Poptip renders correctly 1`] = `
<Tooltip
  animateFill={true}
  animation="shift"
  arrow={true}
  arrowSize="regular"
  className=""
  delay={0}
  disabled={false}
  distance={10}
  duration={375}
  followCursor={false}
  hideDelay={0}
  hideDuration={375}
  hideOnClick={true}
  html={<div />}
  inertia={false}
  interactive={false}
  interactiveBorder={2}
  multiple={false}
  offset={0}
  onHidden={[Function]}
  onHide={[Function]}
  onRequestClose={[Function]}
  onShow={[Function]}
  onShown={[Function]}
  popperOptions={Object {}}
  position="top"
  size="regular"
  sticky={false}
  stickyDuration={200}
  style={Object {}}
  theme="dark"
  touchHold={false}
  trigger="mouseenter focus"
  unmountHTMLWhenHide={false}
>
  <div>
    Testing Poptip
  </div>
</Tooltip>
`;
